src/test/fixtures/profiles/make-profile.js

```diff
@@ -152,182 +152,189 @@ export function getEmptyThread(overrides: ?Object): Thread {
  * is trimmed.
  */
 export function getProfileFromTextSamples(
-  text: string
-): { profile: Profile, funcNames: string[] } {
+  ...textList: string[]
+): { profile: Profile, funcNamesPerThread: Array<string[]> } {
   const nonEmpty = t => t;
-  const lines = text.split('\n').filter(nonEmpty);
+  const profile = getEmptyProfile();
+  const funcNamesPerThread = [];
 
-  // Compute the index of where the columns start in the string. String.prototype.split
```

Comment:
The diff for this file looks really crazy, but really the code is mostly all the same, with a tad bit of re-ordering and and indention changes to repeat the relevant parts to duplicate the code to run on multiple threads.

Reply:
TBH I won't review this thoroughly given it's in tests.
Usually I'm not really happy with complexity in tests... Maybe it would help if you'd split the big function in smaller functions with descriptive names.